lib/rack_graphql/application.rb

```diff
@@ -8,18 +9,30 @@ def self.call(
       log_exception_backtrace: RackGraphql.log_exception_backtrace,
       health_route: true,
       health_response_builder: RackGraphql::HealthResponseBuilder,
-      error_status_code_map: {}
+      error_status_code_map: {},
+      error_response_middleware: RackGraphql::ExceptionJSONResponseMiddleware,
+      middlewares: []
     )
+      # rubocop:enable Metrics/ParameterLists
 
       ::Rack::Builder.new do
+        if error_response_middleware
+          use(
+            error_response_middleware,
+            app_name: app_name,
+            error_status_code_map: error_status_code_map,
+            log_exception_backtrace: log_exception_backtrace
+          )
+        end
+        use(::RackGraphql::LogErrorsMiddleware, logger: logger, log_exception_backtrace: log_exception_backtrace)
+        middlewares.each { |middleware| use middleware }
```

Comment:
The main problem I see with this approach is that even though it's intended to allow injecting sentry middleware, it doesn't really work very well for this use case.

Consider a typical application, which uses a couple of middleware and a graphql app:

```
use Raven::Middleware
use Middleware1
use Middleware2
use Middleware3
run GraphQL::Application.new(...)
```

You put the the error tracking middleware up top, so that it catches any unhandled exception. Even if you allow adding custom middleware within the GQL Application, you still have to have the outer exception tracker to handle anything that can be thrown by the middleware or maybe the GQL application itself. Thus, you are forced to use the sentry middleware twice, and this middleware (at least in the new gem) does more things than just catch errors - it also creates a span, initializes a new hub/client, etc.

The sentry middleware isn't intended to be used twice, and I don't think this is the way to go. What we need is a way to inject `Sentry.capture_exception` somewhere, where the exception is available.